python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```
@@ -50,26 +51,53 @@ def __init__(self, function, args=(), kwargs={},
         self.thread = None
         self.description = ''
 
+def getNumAliveThreads():
+    num_currently_running_command = 0
+    for this_thread in ThreadPool:
+        if this_thread._currently_running_command:
+            num_currently_running_command += 1
+    return num_currently_running_command
+
+def checkHeartBeat():
+
+    latest_timeNow = time.time()
+
+    for this_thread in ThreadPool:
+        thread_name = this_thread._thread_name
+        if heartbeat_times[thread_name] > 0:
+            last_time = heartbeat_times[thread_name]
+        else:
+            last_time = global_start_time
```

defaultdict is essentially a way of hiding all the typing of setdefault :wink: